hs_core/views/__init__.py

```diff
@@ -697,3 +697,18 @@ def _set_resource_sharing_status(request, user, resource, is_public, is_discover
         # set isPublic metadata AVU accordingly
         istorage = IrodsStorage()
         istorage.setAVU(resource.short_id, "isPublic", str(resource.raccess.public))
+
+
+def _get_message_for_setting_resource_flag(resource, resource_flag):
```

Comment:
Agreed returning multiple values does not sound a good solution here. I am not sure whether we need "can_be_public()" in the first place, though. Can we just use two checks,  "has_all_required_elements()" and "has_required_content_files"  and'ed together to represent "can_be_public"? This way you will know which is which and don't have to call individual check again to determine which is missing.